go/multipooler/heartbeat/writer.go

```diff
@@ -136,22 +130,30 @@ func (w *Writer) enableWrites() {
 	}()
 }
 
-// disableWrites deactivates heartbeat writes
+// disableWrites deactivates heartbeat writes.
+// Order of operations:
+//  1. Writer is marked closed (by caller). This prevents new writeHeartbeats from running.
+//  2. Cancel the context for any ongoing write to unblock it.
+//  3. Stop the ticks and wait for any in-flight callback to complete.
+//
+// Context cancellation handles query termination automatically via the connection pool.
 func (w *Writer) disableWrites() {
-	// We stop the ticks in a separate go routine because it can block if the write is stuck on full-sync ACKs.
-	// At the same time we try and kill the write that is in progress. We use the context and its cancellation
-	// for coordination between the two go-routines. In the end we will have guaranteed that the ticks have stopped
-	// and no write is in progress.
-	ctx, cancel := context.WithCancel(context.TODO())
-	go func() {
-		w.ticks.Stop()
-		cancel()
-	}()
-	w.killWritesUntilStopped(ctx)
+	// Cancel any ongoing write to unblock it
+	w.writeMu.Lock()
+	if w.writeCancel != nil {
+		w.writeCancel()
```

Comment:
does this cancel require making a separate connection to Postgres (like it does in lib/pq)? if so, is the pool guaranteed to have a connection available for that?